File: src/services/instructions/loadstore/ST_W_MREGX_XX_REGY.java

 import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `ld.s ${sdestination}, [${ssource}]`

export default class ST_W_MREGX_XX_REGY extends Instruction {
	public ST_W_MREGX_XX_REGY(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setArgument32()
		super.setAssembler("st.w [" + this.sdest + " + 0x%08x], " + this.ssource)
	}
	
	exec ({ context, memory }) {
		setMemContent(context, (fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]]   +  this.argument)) / 2, context[REGISTER_VALUE_NAME_MAPPER[this.source]] , Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]]   +  this.argument))

		context.pc  += 6
		updateViewer32(context, Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]]  + this.argument), context[REGISTER_VALUE_NAME_MAPPER[this.source]] )
	}
}

File: src/services/instructions/loadstore/StWMRegXXXRegY.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `st.w [${sdestination} + 0x%08x], ${ssource}`

export default class StWMRegXXXRegY extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setArgument32(memory)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource))
	}

	exec ({ context }) {
		Instruction.setMemContent(
			context,
			(Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]]   +  this.argument)) / 2,
			context[REGISTER_VALUE_NAME_MAPPER[this.source]],
			Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.destination]]   +  this.argument)
		)

		context.pc  += 6
	}
}
